### PR TITLE
feat: Add new simplified model for profile showcase preferences

### LINKED ISSUE #13688: feat(ProfileShowcase): Simplified NIM backend for profile showcase settings
## Description

This task is backend-side counterpart of https://github.com/status-im/status-desktop/issues/13498

Dirty state is agreed to be handled on the UI side so there is no need to maintain it on Nim side. This make the nim side much simpler.

Nim side should expose read-only models with following roles:

- key
- visibility
- position

For all considered entities:
- communities
- accounts
- collectibles
- assets

**Note**: probably it's also possible to skip `position` role and just assume the order in the model already reflects the desired order

The only exception is `Web` section which should provide all necessary information directly from the model and there is no visibility role.

Saving updated showcase settings should be done via single method taking js/json object of e.g. following structure:
```
{
  accounts: [ { id: <string>, visibility: <int> }],
  communities: [ { id: <string>, visibility: <int> }],
  assets: [ { id: <string>, visibility: <int> }],
  collectibles: [ { id: <string>, visibility: <int> }],
  web: [ { all necessary data here }]
}
```
**Note**: If for some reason it's necessary to provide more info to save method (e.g. bc on nim side it's unhandy to get some metadata), it can be also specified as a required field. It shouldn't be a problem on UI side to provide that info as it's freely available there.

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #13688
Waits https://github.com/status-im/status-go/pull/4854#issuecomment-1973340896

### What does the PR do

- [x] Add simplified profile showcase preferences models for:
  - [x] Communities
  - [x] Accounts
  - [x] Collectibles
  - [x] Assets
  - [x] Social links

-  [x] Add api for saving profile changes 

JSon structure:
```
{
  accounts: [ { key <string>, visibility: <int> }],
  communities: [ { key: <string>, visibility: <int> }],
  assets: [ { key: <string>, visibility: <int> }],
  collectibles: [ { key: <string>, visibility: <int> }],
  socialLinks: [ { url: <string>, text: <string>, visibility: <int> }]
}
```

### Affected areas

Profile showcase

### CODE REVIEW

File: src/app/modules/main/profile_section/profile/module.nim
Comment (MishkaRogachev) [inline src/app/modules/main/profile_section/profile/module.nim:165]: @alexjba @noeliaSD @micieslak 
this part looks dirty from my perspective, to simplify it we need to add some extra data to the initial json: 
- wallet accounts metadata
- collectibles's raw identification fields
-and i expect some extra data for tokens (contractAddress and tokeId)

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (caybro) [inline src/app/modules/main/profile_section/profile/models/profile_save_data.nim:39]: These 3 lines get repeated over and over; some macro/inline proc? :)

File: src/app/modules/main/profile_section/profile/view.nim
Comment (caybro) [inline src/app/modules/main/profile_section/profile/view.nim:48]: Does it make sense to publish the social links related stuff when it's still TODO?

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (caybro) [inline ui/app/AppLayouts/Profile/stores/ProfileStore.qml:112]: Missing handling the `json` param? This is why I hate exposing the strongly typed backend API via a store... You won't even get a warning here

**Review by caybro — CHANGES_REQUESTED** [review]

File: src/app/modules/main/profile_section/profile/view.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/view.nim:48]: Was waiting for https://github.com/status-im/status-go/pull/4775, will update PR soon with social links soon

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml]: fixed

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim]: fixed

**Comment by MishkaRogachev** [comment]
For reviewers: this PR is not for merging for now, it's a proposal

File: src/app/modules/main/profile_section/profile/module.nim
Comment (alexjba) [reply src/app/modules/main/profile_section/profile/module.nim:168]: `the initial json` means the json composed in qml when saving?

I find it ok if this data is really needed here, in `saveProfileShowcasePreferences`. My reasoning is that the profileShowcasePreferences only operates on visibility and position. If the metadata is needed on save operation, I find it ok to be collected on save as opposed to be carried around in other places where it's not immediatly needed.

But I don't really understand why this data needs to be collected here just to be forwarded to status-go. I'd expect status-go not to need accounts metadata for example when saving some accounts showcase preferences. IMO this data could be kept in sync by status-go internals.

Unless I'm missing something, I think it would be more or less the same if we're collecting this data in `saveProfileShowcasePreferences` or qml. The best solution IMO is for this data not to be needed on save. The profile showcase preferences is not meant to save accounts metadata, just some visibilities and positions.

Comment (alexjba) [reply src/app/modules/main/profile_section/profile/module.nim:168]: On the other hand, I see it's quite an overhead to parse the qml updates. The profile settings are going through the following changes:

---qml---
1. model to string
---nim---
2. string -> json
3. json -> nim object 1
4. nim object 1 -> nim object 2 (similar object)
5. nim object 2 -> json
6. json -> back to string

would your proposal help in simplifying this flow as well?

Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/module.nim:168]: @alexjba thanks for the detailed comment, let's try to figure out the best way to do it!

Yes, by json I meant the data that qml passes to nim for saving. 

There is one thing about filling additional fields in status-go: these data are in different services and not all status-go api are available from messenger service. It is still possible to route them, but it is much more laborious than in nim.  However, for profile showcase preferences this problem is not so noticeable. I can move wallet filling to go, and it will even be correct. But not everything is so easy with tokens. The most correct structure seems to me to be:

```
{
  communities: [ { communityId: <string>, visibility: <int> }],
  accounts: [ { address <string>, visibility: <int> }],
  collectibles: [ { chainId: <int>, tokenId: <string>, contractAddress: <string>, visibility: <int> }],
  verifiedTokens: [ { symbol: <string>, visibility: <int> }],
  unverifiedTokens: [ { chainId: <int>, contractAddress: <string>, visibility: <int> }],
  socialLinks: [ { url: <string>, text: <string>, visibility: <int> }]
}
```

I would prefer to use the same nim structure for both saving and querying profile showcase preferences from status-go. And it is possible to completely remove converying in the nim layer, but then the position and correct id names will have to be filled in qml.

File: src/app_service/service/profile/dto/profile_showcase_preferences.nim
Comment (MishkaRogachev) [reply src/app_service/service/profile/dto/profile_showcase_preferences.nim:113]: ill improve it soon, after merging current PRs

File: src/app/modules/main/profile_section/profile/module.nim
Comment (noeliaSD) [reply src/app/modules/main/profile_section/profile/module.nim:168]: For `verifiedTokens` and `unverifiedTokens` I'd use the word `assets` instead of `tokens` to have more nomenclature consistency!

File: src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim
Comment (noeliaSD) [inline src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim:7]: Do we then need the position? I think we talked about using the natural order at least in the UI, with the natural order it should be enough

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (noeliaSD) [inline src/app/modules/main/profile_section/profile/models/profile_save_data.nim:10]:
```suggestion
  visibility*: ProfileShowcaseVisibility
```

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (noeliaSD) [inline src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: Why do we need a different type than `ProfileShowcasePreferencesItem` in here?

File: src/app/modules/main/profile_section/profile/module.nim
Comment (noeliaSD) [reply src/app/modules/main/profile_section/profile/module.nim:168]: > @alexjba thanks for the detailed comment, let's try to figure out the best way to do it!
> 
> Yes, by json I meant the data that qml passes to nim for saving.
> 
> There is one thing about filling additional fields in status-go: these data are in different services and not all status-go api are available from messenger service. It is still possible to route them, but it is much more laborious than in nim. However, for profile showcase preferences this problem is not so noticeable. I can move wallet filling to go, and it will even be correct. But not everything is so easy with tokens. The most correct structure seems to me to be:
> 
> ```
> {
>   communities: [ { communityId: <string>, visibility: <int> }],
>   accounts: [ { address <string>, visibility: <int> }],
>   collectibles: [ { chainId: <int>, tokenId: <string>, contractAddress: <string>, visibility: <int> }],
>   verifiedTokens: [ { symbol: <string>, visibility: <int> }],
>   unverifiedTokens: [ { chainId: <int>, contractAddress: <string>, visibility: <int> }],
>   socialLinks: [ { url: <string>, text: <string>, visibility: <int> }]
> }
> ```
> 
> I would prefer to use the same nim structure for both saving and querying profile showcase preferences from status-go. And it is possible to completely remove converying in the nim layer, but then the position and correct id names will have to be filled in qml.

I'm not sure if I'm understanding correctly what you say here but if I'm not wrong for the UI to do the proper joins we will need to indentify each item by a `key` being this `key` the following corresponding data for each case:
- Communities --> `key` = `communityId`
- Accounts --> `key` = `walletAddress`
- VerifiedTokens (assets) --> `key` = `symbol` in case of _non-community_ ones but a `hash(chainId, tokenId)` or something (we need to look at the `assetsModel` from wallet) for _community ones_ 
- UnverifiedTokens --> I really don't know about how these are treat in wallet
- Collectibles --> `key` = `uid` value build from _chainId_, _tokenId_ and _contractAddress_ 

I think that if there's the need of some `key` transformation, it should happen in `nim` and the UI should not know anything about the rest of params, just the `key`.

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml:34]: Okey, ill update qml & nim parts together

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml:111]: Not for now, because of profile image. Let's open an issue instead :)

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:10]: I would like to stick with [showcaseVisibility](https://github.com/status-im/status-desktop/issues/13688#issuecomment-1960210809)

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: In case we have similar requirements to the model source item and json structure, it may work. But on the latest discussion stage they are a bit different.

File: src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim:7]: If we don't have plans to support diff changes, i can remove it. But just in case we will need it in the future it's worth having them in the model.

**Comment by MishkaRogachev** [comment]
> For verifiedTokens and unverifiedTokens I'd use the word assets instead of tokens to have more nomenclature consistency!

I don't have strong opinion here, but it's better to have same naming as wallet team @dlipicar @alaibe @Khushboo-dev-cpp WDYT?

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml:111]: Will you open it? Thanks!

File: src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim
Comment (noeliaSD) [reply src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim:7]: @alexjba @micieslak wdyt?

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (noeliaSD) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: Which are the differences?

Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: fields naming and extra fields for collectibles

File: src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim
Comment (alexjba) [reply src/app/modules/main/profile_section/profile/models/profile_showcase_preferences_item.nim:7]: I think it's ok to have it in the preferences model. We already have code in the new models that depends on `showcasePosition`. IMO the contacts profile doesn't need the position and can use the natural order.

**Comment by alexjba** [comment]
One note: currently when saving the profile showcase the app is briefly freezing. Could we move the `save` to an async task?

**Comment by MishkaRogachev** [comment]
> One note: currently when saving the profile showcase the app is briefly freezing. Could we move the `save` to an async task?

Yes! I will update the PR soon

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml:111]: https://github.com/status-im/status-desktop/issues/13767

File: ui/app/AppLayouts/Profile/stores/ProfileStore.qml
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Profile/stores/ProfileStore.qml:34]: I would like to stick with `showcasePreferences***Model` to keep naming aligned with backend. And profile showcase for a contact should be located in the contacts module, so i think `own` is extra.

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: Even if they are two identical structures, I'm against using the same type for models and for reflecting the json structure. It would be a time bomb.

File: src/app/modules/main/profile_section/profile/module.nim
Comment (alexjba) [reply src/app/modules/main/profile_section/profile/module.nim:168]: > I think that if there's the need of some key transformation, it should happen in nim and the UI should not know anything about the rest of params, just the key.

I think it's a good point to keep the nim specific logic in nim so that we don't have to duplicate it in qml as well.

File: src/app/modules/main/profile_section/profile/models/profile_save_data.nim
Comment (noeliaSD) [reply src/app/modules/main/profile_section/profile/models/profile_save_data.nim:8]: > Even if they are two identical structures, I'm against using the same type for models and for reflecting the json structure. It would be a time bomb.

Probably I'm missing something but I'd opt for using same type of object when we are talking about the same data instead of creating different types each time we do different actions, otherwise we are acumulating types and types that indeed define the same things. However, you have a better view of the complete flow so I leave it up to you! It was just a suggestion! :)

File: src/app/modules/main/profile_section/profile/models/showcase_preferences_generic_model.nim
Comment (alexjba) [inline src/app/modules/main/profile_section/profile/models/showcase_preferences_generic_model.nim:17]: This model won't have the `showcasePosition`?

File: src/app_service/service/profile/dto/profile_showcase_preferences.nim
Comment (alexjba) [reply src/app_service/service/profile/dto/profile_showcase_preferences.nim:106]: > ill improve it soon, after merging current PRs

After merging this PR?

**Review by alexjba — COMMENTED** [review]
LGTM in general! Nice work!

2 questions inline:

**Comment by MishkaRogachev** [comment]
> After merging this PR?

@alexjba yes, i want refactor this to helpers function and use it in the wallet section too

File: src/app/modules/main/profile_section/profile/models/showcase_preferences_generic_model.nim
Comment (MishkaRogachev) [reply src/app/modules/main/profile_section/profile/models/showcase_preferences_generic_model.nim:17]: i'll add it now

**Review by alexjba — APPROVED** [review]
Nice work! Thanks! 🍻

**Review by noeliaSD — APPROVED** [review]
Tested on https://github.com/status-im/status-desktop/pull/13886

**Review by caybro — APPROVED** [review]

**Comment by alexjba** [comment]
@MishkaRogachev Found an issue while saving the collectibles. The status-go error is saying the owner account is not in showcase, but I see it there.

```
DBG 2024-03-08 18:19:23.005+02:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=32866290 file=threadpool.nim:54 messageType=SaveProfileShowcasePreferencesTaskArg:ObjectType threadid=32866290 task="{\"$type\":\"SaveProfileShowcasePreferencesTaskArg:ObjectType\",\"preferences\":{\"$type\":\"ProfileShowcasePreferencesDto:ObjectType\",\"communities\":[{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x021b9be65e5ca8e2699296b23374f57a1a0a8bf14f53c24f67f1637c123701abf8\",\"showcaseVisibility\":2,\"order\":0},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x023bc3823a823a5c899dadebe73cd030c446fbefc2d035d5c943bbebb8f22125be\",\"showcaseVisibility\":2,\"order\":1},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x0269b18891d3b42ebd047be103d05c7da72969fa4417bf6b5f4abd6b6b872e8207\",\"showcaseVisibility\":2,\"order\":2},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x026b5070c012cafcac1477ce19c45563e177f6b7a08bfda6b90bc9fdacc93cd4e1\",\"showcaseVisibility\":2,\"order\":3}],\"accounts\":[{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0x8b6950bb8a74489a83e6a1281e3aa008f02bf368\",\"showcaseVisibility\":3,\"order\":0},{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0xb5102d8abe09650e5f18248b18c09c27df1765b6\",\"showcaseVisibility\":2,\"order\":1},{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0x4f014b9143c52a9f07b7edf9fd79ca2176532140\",\"showcaseVisibility\":3,\"order\":2}],\"collectibles\":[{\"$type\":\"ProfileShowcaseCollectiblePreference:ObjectType\",\"contractAddress\":\"0x21f6f5cb75e81e5104d890d750270ed6538c50cb\",\"chainId\":10,\"tokenId\":\"2\",\"showcaseVisibility\":3,\"order\":0}],\"verifiedTokens\":[],\"unverifiedTokens\":[],\"socialLinks\":[]},\"vptr\":105553123694336,\"slot\":\"asyncProfileShowcasePreferencesSaved\",\"tptr\":4310183228}"
DBG 2024-03-08 18:19:23.005+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866290 file=core.nim:27 rpc_method=wakuext_setProfileShowcasePreferences
DBG 2024-03-08 18:19:23.005+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866203 file=core.nim:27 rpc_method=accounts_addressWasShown
DBG 2024-03-08 18:19:23.006+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866203 file=core.nim:27 rpc_method=accounts_addressWasShown
ERR 2024-03-08 18:19:23.006+02:00 rpc response error                         topics="rpc" tid=32866290 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\n"
ERR 2024-03-08 18:19:23.006+02:00 error doing rpc request                    topics="rpc" tid=32866290 file=core.nim:40 methodName=wakuext_setProfileShowcasePreferences exception="\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\n"
DBG 2024-03-08 18:19:23.018+02:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=32866289 file=threadpool.nim:54 messageType=QObjectTaskArg:ObjectType threadid=32866289 task="{\"$type\":\"QObjectTaskArg:ObjectType\",\"vptr\":105553123694336,\"slot\":\"asyncProfileShowcasePreferencesLoaded\",\"tptr\":4310184416}"
DBG 2024-03-08 18:19:23.018+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866289 file=core.nim:27 rpc_method=wakuext_getProfileShowcasePreferences
ERR 2024-03-08 18:19:23.018+02:00 Error saving profile showcase preferences  topics="profile-service" tid=32866203 file=service.nim:194 msg="\"\\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\\n\""
```

https://github.com/status-im/status-desktop/assets/47811206/ff5ade15-8805-4b2e-8d92-d53a323ed4c7

**Comment by noeliaSD** [comment]
> @MishkaRogachev Found an issue while saving the collectibles. The status-go error is saying the owner account is not in showcase, but I see it there.
> 
> ```
> DBG 2024-03-08 18:19:23.005+02:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=32866290 file=threadpool.nim:54 messageType=SaveProfileShowcasePreferencesTaskArg:ObjectType threadid=32866290 task="{\"$type\":\"SaveProfileShowcasePreferencesTaskArg:ObjectType\",\"preferences\":{\"$type\":\"ProfileShowcasePreferencesDto:ObjectType\",\"communities\":[{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x021b9be65e5ca8e2699296b23374f57a1a0a8bf14f53c24f67f1637c123701abf8\",\"showcaseVisibility\":2,\"order\":0},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x023bc3823a823a5c899dadebe73cd030c446fbefc2d035d5c943bbebb8f22125be\",\"showcaseVisibility\":2,\"order\":1},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x0269b18891d3b42ebd047be103d05c7da72969fa4417bf6b5f4abd6b6b872e8207\",\"showcaseVisibility\":2,\"order\":2},{\"$type\":\"ProfileShowcaseCommunityPreference:ObjectType\",\"communityId\":\"0x026b5070c012cafcac1477ce19c45563e177f6b7a08bfda6b90bc9fdacc93cd4e1\",\"showcaseVisibility\":2,\"order\":3}],\"accounts\":[{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0x8b6950bb8a74489a83e6a1281e3aa008f02bf368\",\"showcaseVisibility\":3,\"order\":0},{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0xb5102d8abe09650e5f18248b18c09c27df1765b6\",\"showcaseVisibility\":2,\"order\":1},{\"$type\":\"ProfileShowcaseAccountPreference:ObjectType\",\"address\":\"0x4f014b9143c52a9f07b7edf9fd79ca2176532140\",\"showcaseVisibility\":3,\"order\":2}],\"collectibles\":[{\"$type\":\"ProfileShowcaseCollectiblePreference:ObjectType\",\"contractAddress\":\"0x21f6f5cb75e81e5104d890d750270ed6538c50cb\",\"chainId\":10,\"tokenId\":\"2\",\"showcaseVisibility\":3,\"order\":0}],\"verifiedTokens\":[],\"unverifiedTokens\":[],\"socialLinks\":[]},\"vptr\":105553123694336,\"slot\":\"asyncProfileShowcasePreferencesSaved\",\"tptr\":4310183228}"
> DBG 2024-03-08 18:19:23.005+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866290 file=core.nim:27 rpc_method=wakuext_setProfileShowcasePreferences
> DBG 2024-03-08 18:19:23.005+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866203 file=core.nim:27 rpc_method=accounts_addressWasShown
> DBG 2024-03-08 18:19:23.006+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866203 file=core.nim:27 rpc_method=accounts_addressWasShown
> ERR 2024-03-08 18:19:23.006+02:00 rpc response error                         topics="rpc" tid=32866290 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\n"
> ERR 2024-03-08 18:19:23.006+02:00 error doing rpc request                    topics="rpc" tid=32866290 file=core.nim:40 methodName=wakuext_setProfileShowcasePreferences exception="\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\n"
> DBG 2024-03-08 18:19:23.018+02:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=32866289 file=threadpool.nim:54 messageType=QObjectTaskArg:ObjectType threadid=32866289 task="{\"$type\":\"QObjectTaskArg:ObjectType\",\"vptr\":105553123694336,\"slot\":\"asyncProfileShowcasePreferencesLoaded\",\"tptr\":4310184416}"
> DBG 2024-03-08 18:19:23.018+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=32866289 file=core.nim:27 rpc_method=wakuext_getProfileShowcasePreferences
> ERR 2024-03-08 18:19:23.018+02:00 Error saving profile showcase preferences  topics="profile-service" tid=32866203 file=service.nim:194 msg="\"\\nstatus-go error [methodName:wakuext_setProfileShowcasePreferences, code:-32000, message:account holding the collectible is not presented in the profile showcase ]\\n\""
> ```
> 
>  Screen.Recording.2024-03-08.at.18.22.31.mov

We can create task for next week!

**Pull request merged**
